### ISSUE TITLE
Expose the cas_id in binary protocol.

### ISSUE THREAD

**Issue by letmutx** [aisk/rust-memcache#80]
This will be useful for letmutx/ghetto-lock-rs#10

**Comment by letmutx** [comment]
Thank coo

**Comment by letmutx** [comment]
One thought is that we could have a `BinaryClient` which works only with binary protocol and have methods that would take `cas_id` for all operations as necessary. @aisk thoughts?

**Comment by aisk** [comment]
I think it's cool to have this method, and for ASCII protocol, seems we can using the mget related API to implement it?

https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L472

**Comment by aisk** [comment]
Sorry maybe I had some misunderstanding, what you want is a single function to get a `cas_id` from a key without the total value body? Or something else?

**Comment by letmutx** [comment]
Yes, metaget and metaset seem to be similar to the ASCII commands to the binary commands I was talking about.

Implementing the feature I want in the binary protocol is much less work with the current code we have, though.
